lib/genome/importers/dtc/dtc.rb

```diff
@@ -0,0 +1,62 @@
+require 'genome/online_updater'
+
+module Genome; module Importers; module Dtc;
+class DtcImporter < Genome::OnlineUpdater
+  attr_reader :file_path, :source
+
+  def initialize(file_path)
+    @file_path = file_path
+  end
+
+  def get_version
+    source_db_version = Date.today.strftime("%d-%B-%Y")
+    @new_version = source_db_version
+  end
+
+  def import
+    remove_existing_source
+    create_new_source
+    create_interaction_claims
+  end
+
+  private
+  def remove_existing_source
+    Utils::Database.delete_source('DTC')
+  end
+
+  def create_new_source
+    @source ||= DataModel::Source.create(
+        {
+            base_url: 'https://drugtargetcommons.fimm.fi/',
+            site_url: 'https://drugtargetcommons.fimm.fi/',
+            citation: 'Drug Target Commons 2.0: a community platform for systematic analysis of drug–target interaction profiles. Tanoli Z, Alam Z, Vähä-Koskela M, Malyutina A, Jaiswal A, Tang J, Wennerberg K, Aittokallio T. Database. 2018.',
+            source_db_version:  get_version,
+            source_type_id: DataModel::SourceType.INTERACTION,
+            source_db_name: 'DTC',
+            full_name: 'Drug Target Commons',
+            license: 'Creative Commons Attribution-NonCommercial 3.0 (BY-NC)',
+            license_url: 'https://academic.oup.com/database/article/doi/10.1093/database/bay083/5096727',
+        }
+    )
+  end
+
+  def create_interaction_claims
+    CSV.foreach(file_path, :headers => true) do |row|
+      next if row['compound_id'].nil? || row['gene_names'].nil?
+      drug_id = row['compound_id']
+      drug_name = row['compound_name']
+      drug_claim = create_drug_claim(drug_id, drug_id, 'CHEMBL ID')
+      if drug_name.present?
+        create_drug_claim_alias(drug_claim, drug_name, 'DTC drug name')
+      gene_name = row['gene_names']
```

Comment:
The `gene_names` column looks like it can contain more than one gene (separated by `, `). So you'll want to split on that string and iterate over them and create one gene claim and interaction claim for each.

Reply:
Isn't that field being separated by `,` going to be an issue since the file itself is comma delimited?

Reply:
Wherever a comma is used within a field, the field is encased by `""` in the plain text file so the CSV parser will know to interpret this as one field and ignore the commas within the field, e.g.:
```
CHEMBL474208,AAAHTCZONLHYSA-UHFFFAOYSA-N,,,"Q96DB2, Q9UBN7, Q92769, Q13547, Q9UKV0, Q9UQL6, Q8WUI4, P56524, O15379, Q969S8, Q9BY41",HISTONE DEACETYLASE,"HDAC1, HDAC10, HDAC11, HDAC2, HDAC3, HDAC4, HDAC5, HDAC6, HDAC7, HDAC8, HDAC9",,,19301902,EC50,=,1740,NM,,"",cell_based,"","","","",HeLa,,,,,,,Inhibition of HDAC in human HeLa cells assessed as induction of cellular histone H3 hyperacetylation,"Design of chimeric histone deacetylase- and tyrosine kinase-inhibitors: a series of imatinib hybrides as potent inhibitors of wild-type and mutant BCR-ABL, PDGF-Rbeta, and histone deacetylases.",J. Med. Chem.,PUBLICATION,
```
notice how `"Q96DB2, Q9UBN7, Q92769, Q13547, Q9UKV0, Q9UQL6, Q8WUI4, P56524, O15379, Q969S8, Q9BY41"` and `"HDAC1, HDAC10, HDAC11, HDAC2, HDAC3, HDAC4, HDAC5, HDAC6, HDAC7, HDAC8, HDAC9"` each are encased in double quotes so it will be parsed as a single field. This is controlled by the `quote_char` option in the CSV reader and is by default set to `"`. See https://ruby-doc.org/stdlib-2.6.1/libdoc/csv/rdoc/CSV.html for more info.